### ISSUE TITLE
Framework does not build with rust 1.61

### ISSUE THREAD

**Issue by sylvainpelissier** [neodyme-labs/solana-poc-framework#9]
To reproduce:
```bash
$ rustup default 1.61 
$ cargo build                                                                                                                             ✔ 
   Compiling libsecp256k1 v0.5.0
   ...
   Compiling poc-framework v0.1.6 (/home/sylvain/software/solana-poc-framework)
error: expected None-delimited group
   --> src/lib.rs:741:21
    |
741 |                     solana_bpf_loader_upgradeable_program!(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `solana_bpf_loader_upgradeable_program` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected None-delimited group
   --> src/lib.rs:742:21
    |
742 |                     solana_bpf_loader_program!(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `solana_bpf_loader_program` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected None-delimited group
   --> src/lib.rs:743:21
    |
743 |                     solana_bpf_loader_deprecated_program!(),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the macro `solana_bpf_loader_deprecated_program` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `poc-framework` due to 3 previous errors
```

It seems to be related to https://github.com/rust-lang/rust/pull/92472. It is solved by switching to rust 1.60:
```bash
$ rustup default 1.60
```

**Comment by tlambertz** [comment]
This is caused by (old) Solana upstream not building with 1.60 anymore.

I think their fix is in https://github.com/solana-labs/solana/pull/24390

We'll have to update the core solana dependency of the poc-framework, but that'll take some work as we interact with internals that have changed now.

The easiest fix for now is to use an older compiler.

**Comment by riptl** [comment]
Fixed in #13

**Comment by riptl** [comment]
This can be closed @CherryWorm
